src/crypto/ecdh.rs

```diff
@@ -124,14 +126,20 @@ impl Decryptor for SecretKey {
 
                 shared_secret.to_bytes().to_vec()
             }
-            SecretKey::P256 { secret, .. } => {
-                derive_shared_secret_decryption::<p256::NistP256>(public_point, secret, &curve, 65)?
-            }
-            SecretKey::P384 { secret, .. } => {
-                derive_shared_secret_decryption::<p384::NistP384>(public_point, secret, &curve, 97)?
-            }
+            SecretKey::P256 { secret, .. } => derive_shared_secret_decryption::<p256::NistP256>(
+                data.public_point,
+                secret,
+                &curve,
+                65,
```

Comment:
maybed make these constants

Reply:
Agreed, but I'd like to do that in a separate PR, to keep this PR somewhat focused.
I opened #402 to track our intention to improve the constants situation.